akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -620,6 +620,22 @@ private[akka] class RemoteActorRefProvider(
         local.addressString
     }
   }
+
+  override lazy val systemUid: Long = {
+    val address = getDefaultAddress
+    // 12 bits from the host
+    val hostPart = address.host.getOrElse("").hashCode & ((1 << 12) - 1)
+    // 16 bits from port, all of max 65535
+    val portPart = address.port.getOrElse(0).toLong
+    // 36 bits from the time, which corresponds to 795 days
+    val timePart = System.currentTimeMillis() & ((1L << 36) - 1)
```

Comment:
Not sure it's the right ratio of bits for these parts.

Thinking some scenarios.
1. Two ActorSystems are started at the same time from the same host. Then the port will make them different.
2. Two ActorSystems are started at the same time from different host and same port. Then the host part might be too small? This is probably rather common for a Kubernetes cluster.
3. An ActorSystem is restarted with same host and port as previous incarnation. The time part makes it unlikely to become the same uid.

Reply:
Maybe break this out into a static method and do some looping test with a set of host/ports and see how/if/when it causes conflict?